### PR TITLE
Fix #13451 - make `afs` work ##anal

### LINKED ISSUE #13451: Fix `afs` to use `td`/`tl`
Currently `afs` is just a stub. To think how better to implement it - either make an alias for `td` or something else.

### PR DESCRIPTION
**Pull request by XVilka**
Also addressing this bug: https://github.com/radare/radare2/issues/13507

Corresponding R2R PR: https://github.com/radare/radare2-regressions/pull/1763

### CODE REVIEW

File: libr/anal/meson.build [hunk 1]
```diff
@@ -133,6 +134,7 @@ r_anal = library('r_anal', r_anal_sources,
     r_crypto_dep,
     r_search_dep,
     r_cons_dep,
+	r_parse_dep,
```
Comment (pelijah) [inline libr/anal/meson.build]: 2 spaces

File: libr/anal/meson.build [hunk 2]
```diff
@@ -158,6 +160,7 @@ pkgconfig_mod.generate(r_anal,
     'r_reg',
     'r_asm',
     'r_crypto',
+	'r_parse',
```
Comment (pelijah) [inline libr/anal/meson.build]: Same here

File: libr/anal/fcn.c [hunk 3]
```diff
@@ -2191,29 +2192,19 @@ R_API char *r_anal_fcn_to_string(RAnal *a, RAnalFunction *fs) {
...
+		r_anal_save_parsed_type (a, out);
 	}
-	length = strlen (sig) + 10;
-	/* Add 'function' keyword */
-	char *str = calloc (1, length);
-	if (!str) {
-		eprintf ("Cannot allocate %d byte(s)\n", length);
-		return false;
+	if (error_msg) {
+		fprintf (stderr, "%s", error_msg);
```
Comment (radare) [inline libr/anal/fcn.c]: Use eprintf

File: libr/anal/type.c [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	}
+}
+
+R_API void r_anal_save_parsed_type(RAnal *anal, const char *parsed) {
+	r_return_if_fail (anal && parsed);
+	// First, if this exists, let's remove it.
+	char *type = strdup (parsed);
+	if (type) {
+		char *name = NULL;
+		if ((name = strstr (type, "=type")) ||
```
Comment (radare) [inline libr/anal/type.c]: Move into separate function. This seems omething we can simplify if we prefix al the tkeywords with $ or so

File: libr/anal/type.c [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+			(name = strstr (type, "=enum")) ||
+			(name = strstr (type, "=typedef")) ||
+			(name = strstr (type, "=func"))) {
+			*name = 0;
+			while (name - 1 >= type && *(name - 1) != '\n') {
+				name--;
+			}
+		}
+		if (name) {
+			const char *cname = strdup (name);
```
Comment (radare) [inline libr/anal/type.c]: Strdup doesnt returns const

File: libr/core/cmd_print.c [hunk 6]
```diff
@@ -1207,7 +1207,7 @@ static void cmd_print_format(RCore *core, const char *_input, const ut8* block,
 				char *error_msg = NULL;
 				char *out = r_parse_c_file (core->anal, path, &error_msg);
 				if (out) {
-					r_core_save_parsed_type (core, out);
+					r_anal_save_parsed_type (core->anal, out);
```
Comment (radare) [inline libr/core/cmd_print.c]: Good refactor

File: libr/libs.mk [hunk 7]
```diff
@@ -4,7 +4,7 @@ ifeq ($(LIBS0),)
 LIBS0=util
 LIBS1=socket hash reg cons magic bp search config
 LIBS2=syscall lang io crypto flag
-LIBS3=fs anal bin parse
+LIBS3=fs parse anal bin
```
Comment (radare) [inline libr/libs.mk]: Those 4 libs are built in paralel. If anal depends on parse must be moved into libs4

**Pull request merged**